packages/driver/cypress/integration/commands/querying_spec.js

```diff
@@ -357,6 +357,38 @@ describe('src/cy/commands/querying', () => {
       })
     })
 
+    describe('shadow dom', () => {
+      beforeEach(() => {
+        cy.visit('/fixtures/shadow-dom.html')
+      })
+
+      it('finds element within shadow dom with includeShadowDom option', () => {
+        cy.get('#parent-of-shadow-container-0').within(() => {
+          cy
+          .get('p', { includeShadowDom: true })
+          .should('have.length', 1)
+          .should('have.text', 'Shadow Content 3')
+        })
+      })
+
+      it('when within subject is shadow root, finds element without needing includeShadowDom option', () => {
+        cy.get('#shadow-element-1').shadow().within(() => {
+          cy
+          .get('p')
+          .should('have.length', 1)
+          .should('have.text', 'Shadow Content 1')
+        })
+      })
+
+      it('when within subject is already in shadow dom, finds element without needing includeShadowDom option', () => {
```

Comment:
so... does this mean that if i have somethin like this (apologies if this is incorrect shadow DOM syntax, i hope my meaning gets across):

```
<shadow-root>
	<foo>
		<shadow-root>
			<bar/>
        </shadow-root>
	</foo>
</shadow-root>
```

...and I do this:

```
cy.get('foo', { includeShadowDom: true }).within(() => {
 	cy.get('bar')
})
```

...`<bar/>` will successfully be found, because `foo` is within a shadow DOM already? even though it traverses an additional shadow bounary?

Reply:
No, you still need to pass the option to get `bar` without enabling it at a higher level (test config, cypress.json, etc). The option is only in effect for the duration of the command.

So `cy.get('bar', { includeShadowDom: true })` outside the `within` could traverse through multiple shadow boundaries and find `bar`. However, `cy.within` does not persist the `includeShadowDom` behavior. It acts as it normally does and scopes queries inside so if there was another `bar` outside `foo`, it would not also be in the result of `cy.get('bar')`.

Reply:
ok, that seems reasonable, thanks